src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/Collection.vb

```diff
@@ -0,0 +1,832 @@
+' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Globalization
+Imports System.Diagnostics
+Imports System.Runtime.Serialization
+Imports System.Security
+Imports System.Security.Permissions
+Imports Microsoft.VisualBasic.CompilerServices
+Imports Microsoft.VisualBasic.CompilerServices.ExceptionUtils
+Imports Microsoft.VisualBasic.CompilerServices.Utils
+
+Namespace Microsoft.VisualBasic
+
+    <Serializable()>
```

Comment:
In .NET Core, we are being selective with use of `[Serializable]`. We have limited it to only types where it is definitely needed. This is because of the maintenance burden and historical security issues related to binary serialization.

This type may or may not deserve `[Serializable]`. @ViktorHofer do we have data on how commonly this is seen serialized? 

Here's a list of the types we DO support it on: https://docs.microsoft.com/en-us/dotnet/standard/serialization/binary-serialization

When we do support it, we add special comments to deter anyone renaming private fields (if it does memberwise serialization) like this https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Guid.cs#L25
Also we add special tests for it.

Reply:
@bbowyersmyth I suggest pulling it out for this PR (the attribute, the constructor and the ISerializable implementation). It would need more tests anyway. If we get feedback, we can discuss putting it back. It looks like the only other types marked serializable in this assembly are exceptions, plus `StaticLocalInitFlag`. We have consistently tried to enable serialization on exceptions, so we should probably leave it on those, and add the special tests for them. But again initial PR's could go in without it.

Otherwise change LGTM

Reply:
I strongly recommend not adding more types to the serialization bucket unless there is strong customer demand for it. I don't think we have meaningful data which would allow us making a profound decision for VB types.

Reply:
Removed